compiler/cbmc/src/goto_program/expr.rs

```diff
@@ -815,7 +815,7 @@ impl Expr {
                     && (lhs.typ.is_pointer() || lhs.typ.is_numeric() || lhs.typ.is_vector())
                     || (lhs.typ.is_pointer() && rhs.typ.is_integer())
             }
-            Plus => {
+            Plus | OverflowMinus | OverflowMult | OverflowPlus => {
```

Comment:
The argument types have to remain the same for each operation (i.e., `OverflowMinus` goes with `Minus`).

Reply:
Makes sense. Do you know why the minus has a different retirement?

Reply:
Yes, minus allows both operands to be pointers because it is valid to compute offsets between them. This does not apply to plus because adding two pointers will likely result in another pointer to who-knows-where.